runtime/binding-kafka/src/main/java/io/aklivity/zilla/runtime/binding/kafka/internal/stream/KafkaMergedFactory.java

```diff
@@ -2070,6 +2071,14 @@ private void onProducePartitionLeaderError(
                     produceStreams.remove(leader);
                 }
             }
+            else if (error == ERROR_INVALID_RECORD)
+            {
+                doMergedInitialReset(traceId, ex -> ex.set((b, o, l) -> kafkaResetExRW.wrap(b, o, l)
+                        .typeId(kafkaTypeId)
+                        .error(error)
+                        .build()
+                        .sizeof()));
```

Comment:
This is a change of behavior on error, such that the same checks in `doMergedInitialResetIfNecessary()` are being bypassed when we call `doMergedInitialReset()` directly.